src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -203,22 +179,79 @@ def apply_mixins(cls, data: Dict) -> Dict:
         if "mixins" not in data or "entities" not in data:
             return data
 
-        for entity in data["entities"].values():
+        entities = data["entities"]
+        for entity_name, entity in entities.items():
             entity_mixins = entity_mixins_to_list(
                 entity.get("meta", {}).get("use_mixins")
             )
 
-            entity_fields = get_allowed_fields_for_entity(entity)
-            if entity_fields and entity_mixins:
-                for mixin_name in entity_mixins:
-                    if mixin_name in data["mixins"]:
-                        for key, value in data["mixins"][mixin_name].items():
-                            if key in entity_fields:
-                                entity[key] = value
-                    else:
-                        raise ValueError(f"Mixin {mixin_name} not found in mixins")
+            merged_values = cls._merge_mixins_with_entity(
+                entity_name=entity_name,
+                entity=entity,
+                entity_mixins_names=entity_mixins,
+                mixins=data["mixins"],
+            )
+            entities[entity_name] = merged_values
+        return data
+
+    @staticmethod
+    def _merge_mixins_with_entity(
+        entity_name: str, entity: dict, entity_mixins_names: list, mixins: dict
```

Comment:
The arg names are pretty ambiguous, WDYT of this suggestion?
```suggestion
        entity_name: str, entity: dict, entity_mixins_names: list, mixin_defs: dict
```